pandas/sparse/array.py

```diff
@@ -600,20 +601,43 @@ def convert_sparse(x, axis):
             x = np.atleast_2d(x)
         return x
 
-    typs = com.get_dtype_kinds(to_concat)
-
-    # we have more than one type here, so densify and regular concat
+    if typs is None:
+        typs = com.get_dtype_kinds(to_concat)
+
+    if len(typs) == 1:
+        fill_values = set(c.fill_value for c in to_concat)
+        if len(fill_values) == 1:
+            sp_values = [c.sp_values for c in to_concat]
```

Comment:
can you make a comment here that explains what is happening (that you need 1 typ and 1 fill value to get back a Sparse result), otherwise its dense)